### PR TITLE
fix: js exception when reloading async chunk (#444)

### PR DESCRIPTION
**Pull request by Rulexec**
This PR contains a:

- [x] **bugfix**
- [ ] new **feature**
- [ ] **code refactor**
- [ ] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case

HMR throws exception every time when I do edits in my css files:

```
Uncaught TypeError: Cannot read property 'some' of null
    at getReloadUrl (hotModuleReplacement.js:129)
    at hotModuleReplacement.js:145
    at NodeList.forEach (<anonymous>)
    at reloadStyle (hotModuleReplacement.js:140)
    at update (hotModuleReplacement.js:194)
    at functionCall (hotModuleReplacement.js:24)
```

This fix ignores such obviously wrong updates and allows to use `options.reloadAll`, because this exception happens just before `reloadAll()` call.

Issue — #444 

### Breaking Changes

None.

### CODE REVIEW

**Comment by evilebottnawi** [comment]
Or please create reproducible test repo

**Comment by Rulexec** [comment]
Reproducible [test branch](https://github.com/Rulexec/mini-css-extract-plugin/tree/hmr-exception-demo).

I will try to write test, but it can take a while to investigate, how to write such test.

File: src/hmr/hotModuleReplacement.js [hunk 1]
```diff
@@ -227,3 +231,8 @@ module.exports = function(moduleId, options) {
 
   return debounce(update, 50);
 };
+
+// Used from tests
+module.exports.clearCache = function() {
+  srcByModuleId = Object.create(null);
+};
```
Comment (evilebottnawi) [inline src/hmr/hotModuleReplacement.js:238]: We should use mock for test it, no need it in runtime code

Comment (Rulexec) [reply src/hmr/hotModuleReplacement.js:238]: How to mock variable inside module?

Currently value inside `srcByModuleId = Object.create(null)` persists between tests and my situation happens only if it's clear.

**Pull request closed without merging**